packages/rum-core/src/domain/view/viewMetrics/trackLoadingTime.ts

```diff
@@ -16,9 +17,15 @@ export function trackLoadingTime(
   let isWaitingForLoadEvent = loadType === ViewLoadingType.INITIAL_LOAD
   let isWaitingForActivityLoadingTime = true
   const loadingTimeCandidates: Duration[] = []
+  const firstHidden = trackFirstHidden(configuration)
 
   function invokeCallbackIfAllCandidatesAreReceived() {
-    if (!isWaitingForActivityLoadingTime && !isWaitingForLoadEvent && loadingTimeCandidates.length > 0) {
+    if (
+      firstHidden.timeStamp !== 0 &&
+      !isWaitingForActivityLoadingTime &&
+      !isWaitingForLoadEvent &&
+      loadingTimeCandidates.length > 0
+    ) {
       callback(Math.max(...loadingTimeCandidates) as Duration)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Shouldn't we check the loading time against the firstHidden timestamp like we do with firstContentFullPaint?
Something like
```suggestion
     const loadingTime = Math.max(...loadingTimeCandidates)
      if (loadingTime < firstHidden.timeStamp) {
            callback(loadingTime as Duration)
      }
```

Reply:
Indeed, that was my first approach which will work in this use-case, and also when loading a page then putting it in background before the loading is completed.

Reply:
Will test this case and do the change if needed